lib/ClangImporter/ClangImporter.cpp

```diff
@@ -5213,9 +5213,10 @@ ClangTypeEscapability::evaluate(Evaluator &evaluator,
           } else
             nonPackArgs.push_back(arg);
           for (auto nonPackArg : nonPackArgs) {
-            if (nonPackArg.getKind() != clang::TemplateArgument::Type) {
-              desc.impl.diagnose(loc, diag::type_template_parameter_expected,
-                                 argToCheck.second);
+            if (nonPackArg.getKind() != clang::TemplateArgument::Type &&
+                desc.impl) {
+              desc.impl->diagnose(loc, diag::type_template_parameter_expected,
```

Comment:
If we're okay with disabling these diagnostics when we don't have the `impl` on hand, should we just remove them entirely?

Reply:
I think we should refactor this code at some point. When we import these types we will always have the `impl` at hand. After they are imported, we no longer need to emit diagnostics about these types so we don't really need the `impl`. That being said a better design would be to return some information about the diagnostics that needs to be emitted (instead of having a side effect) and only consume those in the importer and ignore them in other parts of the compiler.